File: packages/zynth-file-intents/android/ZynthFileIntents/src/main/java/dev/zynth/fileintents/ZynthFileIntents.kt
package dev.zynth.fileintents

import android.util.Log
import androidx.activity.ComponentActivity
import androidx.activity.result.contract.ActivityResultContracts
import com.zynth.kit.runtime.ZynthRuntime

object ZynthFileIntents {
    private const val TAG = "ZynthFileIntents"

    @JvmStatic
    fun initialize(activity: ComponentActivity, runtime: ZynthRuntime) {
        Log.d(TAG, "Initializing ZynthFileIntents")

        val module = FileIntentsModule(runtime, activity)
        module.exportLauncher =
            activity.registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
                module.onExportResult(result)
            }

        runtime.installModules(listOf(module))
        Log.d(TAG, "ZynthFileIntents initialized")
    }
}

File: packages/zynth-file-intents/ios/ZynthFileIntents.swift
import Foundation
import ZynthKit

@objc(ZynthFileIntents)
public class ZynthFileIntents: NSObject {
  private static var moduleInstance: FileIntentsModule?

  @objc public static func initialize(with runtime: ZynthRuntime) {
    guard moduleInstance == nil else {
      print("[ZynthFileIntents] Module already initialized")
      return
    }

    let module = FileIntentsModule(runtime: runtime)
    runtime.installModules([module])
    moduleInstance = module
    print("[ZynthFileIntents] Module initialized")
  }

  @objc public static func cleanup() {
    moduleInstance = nil
  }
}
